inference-engine/include/cpp/ie_cnn_network.h

```diff
@@ -241,6 +241,17 @@ class INFERENCE_ENGINE_API_CLASS(CNNNetwork) {
         CALL_STATUS_FNC(serialize, xmlPath, binPath);
     }
 
+protected:
+    /**
+     * @brief Sets weights blob pointer
+     *
+     * @param weights blob pointer
+     */
+    void setWeightsBlobPtr(const Blob::CPtr& weights) {
```

Comment:
Why do we need this method?

Reply:
Need somehow to save a pointer to blob. Lifetime of blob should be the same as for CNNNetwork. 
Idea here  is to read model data   to blob, provide a pointer  to Constant nodes and make them wrap already allocated memory instead of creating new buffers. 
I can do the same as for Blob Proxy and make blob consumers keep pointers, but in this case i would need to expose InferenceEngine types and interfaces to nGraph. Personally i don't like this idea as it will make nGraph dependent on IE, so i;m trying to keep them saparate.

Reply:
BTW, I don't see where this method is used. We should not touch public API for internal needs..

Reply:
@Drizshko could you provide example or test which will show why we need this method and friend class V10Parser here?

Reply:
@ilyachur @ilya-lavrenov i did change implementation, now nGraph buffer itself will store smart pointer to Blob. 
Did this without linking nGraph to IE, by using template parameter.
Please take a look.